editor/gui/scene_tree_editor.cpp

```diff
@@ -209,6 +209,7 @@ void SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent) {
 	}
 
 	TreeItem *item = tree->create_item(p_parent);
+	node_list.insert(item, p_node);
```

Comment:
You also need to clear the list when Tree is cleared.

Reply:
Should clearing the list in _update_tree be enough?

Reply:
Yes, that the only place where `tree` is cleared.